dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -2112,27 +2114,410 @@ than a dMY_CTX in another source file.
+destroyed at any time by another thread executing this function or a
+related one.
+
+Some platforms may allocate the buffer per-thread, so it is thread-safe,
+but you have to use the value (or copy it) before the next call.
+
+There may be a re-entrant function available in which the caller
+furnishes the buffer, and which is thread-safe.  Other flags on the
+entry should indicate this.
+
+=item E
+
+This function accesses the program's environment.
+
+Writing to the environment can create a race with other threads either
+reading or writing the environment at the same time.  If all writers (as
+the core does) protect their accesses by surrounding them with
+
+ ENV_LOCK
+ ...
+ ENV_UNLOCK
+
+then the program is safe with regards to these.  If this is done, then
+read-only accesses can be protected by instead using
+
+ ENV_READ_LOCK
+ ...
+ ENV_READ_UNLOCK
+
+calls surrounding them, which allow for any number of readers at the
+same time.  (A few platforms aren't safe to have multiple concurrent
+readers; for these Perl automatically substitutes an exclusive lock when
+you specify C<ENV_READ_LOCK>, so you don't have to worry about it.)
+
+=item C<L>
+
+This function may read the program's locale.  It should be locale
+thread-safe on systems with thread-safe locales.
+You can wrap it within calls to
+
+ LOCALE_READ_LOCK
+ ...
+ LOCALE_READ_UNLOCK
+
+to make it thread-safe with respect to locale changes by the core in all
+platforms.
+
+If the C<E> flag is also present, you can use
+
+ ENVr_LOCALEr_LOCK
+ ...
+ ENVr_LOCALEr_UNLOCK
+
+or C<ENVw_LOCALEr_LOCK> ...  C<ENVw_LOCALEr_UNLOCK> if the environment
+needs to be locked for exclusive access.
+
+The functions related to formatting likely will not access the locale
+unless there is a C<%> format that is locale-dependent, such as the
+decimal point in floating point conversions, or thousands separators in
+most numeric ones.
+
+=item C<G>
+
+This function accesses global variables, which may be per-thread on
+the platform, but are more likely process-wide.  That means that another
+thread may overwrite those variables before you get a chance to use
+them.  To overcome this, use an exclusive lock, unlocking only after you
+copy to safety the contents of all the variables you need.
+
+Ideally, you would create a mutex for these, but you may be able to use
+the C<ENV_LOCK> for this purpose, provided that using it doesn't unduly
+slow down actual environment accesses.
+
+=item C<Ꞧ>
+
+This function has some sort of race.  Generally this is due to acessing
+some sort of global buffer or state internally
+
+=item C<t>
+
+These variables must not be accessed and functions not called when
+another thread is calling C<tzset>.  Therefore wrap them using
+C<ENVr_LOCALEr_LOCK>.
+
+=item C<T>
+
+These time-related functions may do writes; perhaps through an indirect
+call to C<tzset>, which writes to time-related global variables.
+Therefore they require an exclusive lock.  See C</L</M>>.
```

Comment:
Is the first `/` in `C</L</M>>` intended here?  It displays like:
![image](https://github.com/Perl/perl5/assets/166429/19893c11-9bc9-4eee-a263-cc9b7fc7965e)